agent/engine/execcmdagent/manager_linux_test.go

```diff
@@ -0,0 +1,363 @@
+// +build linux,unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package execcmdagent
+
+import (
+	"context"
+	"errors"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	errors2 "github.com/aws/amazon-ecs-agent/agent/api/errors"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient"
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	mock_dockerapi "github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi/mocks"
+
+	"github.com/docker/docker/api/types"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStart(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_dockerapi.NewMockDockerClient(ctrl)
+	const (
+		testPid1               = 9876
+		testContainerRuntimeId = "123abc"
+		testDockerExecId       = "mockDockerExecID"
+	)
+
+	var (
+		mockError      = errors.New("mock error")
+		mockStartError = StartError{error: mockError, retryable: true}
+		testContainers = []*apicontainer.Container{{
+			RuntimeID: testContainerRuntimeId,
+		}}
+	)
+
+	tt := []struct {
+		execEnabled                bool
+		containers                 []*apicontainer.Container
+		expectCreateContainerExec  bool
+		createContainerExecRes     *types.IDResponse
+		createContainerExecErr     error
+		expectStartContainerExec   bool
+		startContainerExecErr      error
+		expectInspectContainerExec bool
+		inspectContainerExecRes    *types.ContainerExecInspect
+		inspectContainerExecErr    error
+		expectedError              error
+	}{
+		{
```

Comment:
This test is great.  I love that I can see the lifecycle of the start process directly from these test cases.